dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackedEntityAttributeService.java

```diff
@@ -170,64 +181,36 @@ public TrackedEntityAttribute getTrackedEntityAttribute( String uid )
 
     @Override
     @Transactional(readOnly = true)
-    public String validateScope( TrackedEntityAttribute trackedEntityAttribute,
-        String value, TrackedEntityInstance trackedEntityInstance, OrganisationUnit organisationUnit, Program program )
+    public String validateAttributeUniquenessWithinScope( TrackedEntityAttribute trackedEntityAttribute,
+        String value, TrackedEntityInstance trackedEntityInstance, OrganisationUnit organisationUnit )
     {
         Assert.notNull( trackedEntityAttribute, "tracked entity attribute is required." );
+        Assert.notNull( value, "tracked entity attribute value is required." );
 
-        if ( !trackedEntityAttribute.isUnique() || value == null )
+        if ( trackedEntityAttribute.getProgramScope() )
         {
-            return null;
+            log.error( "TrackedEntityAttribute '" + trackedEntityAttribute.getUid() + "' has a programScope. ProgramScope feature is not supported." );
+            return "Attribute " + trackedEntityAttribute.getUid() + " has a programScope. ProgramScope is not supported.";

```

Comment:
Not sure what is most appropriate here. Maybe a warning and not returning? If you add extra, unsupported, params to other things, we dont fail the operation, we just ignore extras... What do you think?

Reply:
OK. Ideal is to remove programScope from the system completely. As it is not supported. Not decided about that yet.

Reply:
So its present in API but not in UI. It seems to me from a glance on existing code, that we would include programscope if it was set? Can you do the same here?

Reply:
Yes, we have it in API and in Database, but not in UI. It is not supported in UI and it probably never will. Because the capture app needs to know the program when registering TEI. And it does not know the program. So having and TEA with programScope is simply IllegalState. It should never happen as it is not supported.